code/modules/food_and_drinks/food/pizza.dm

```diff
@@ -0,0 +1,361 @@
+
+/obj/item/food/pizza/meat
+	name = "meatpizza"
+	desc = "Greasy pizza with delicious meat."
+	icon_state = "meatpizza"
+	food_reagents = list(/datum/reagent/consumable/nutriment = 25, /datum/reagent/consumable/tomatojuice = 6, /datum/reagent/consumable/nutriment/vitamin = 8)
+	foodtypes = GRAIN | VEGETABLES| DAIRY | MEAT
+	slice_type = /obj/item/food/pizzaslice/meat
+
+/obj/item/food/pizza/meat/raw
+	name = "raw meatpizza"
+	icon_state = "meatpizza_raw"
+	foodtypes =  GRAIN | VEGETABLES| DAIRY | MEAT | RAW
+	burns_in_oven = FALSE
+	slice_type = null
+
+/obj/item/food/pizza/meat/raw/MakeBakeable()
+	AddComponent(/datum/component/bakeable, /obj/item/food/pizza/meat, rand(70 SECONDS, 80 SECONDS), TRUE, TRUE)
+
+/obj/item/food/pizza/meat/Initialize()
+	. = ..()
+	AddElement(/datum/element/processable, TOOL_KNIFE, /obj/item/food/pizzaslice/meat, 6, 30)
+
+/obj/item/food/pizzaslice/meat
+	name = "meatpizza slice"
+	desc = "A nutritious slice of meatpizza."
+	icon_state = "meatpizzaslice"
+	tastes = list("crust" = 1, "tomato" = 1, "cheese" = 1, "meat" = 1)
+	foodtypes = GRAIN | VEGETABLES | DAIRY | MEAT
+
+/obj/item/food/pizza/mushroom
+	name = "mushroom pizza"
+	desc = "Very special pizza."
+	icon_state = "mushroompizza"
+	food_reagents = list(/datum/reagent/consumable/nutriment = 28,  /datum/reagent/consumable/nutriment/vitamin = 5)
+	tastes = list("crust" = 1, "tomato" = 1, "cheese" = 1, "mushroom" = 1)
+	foodtypes = GRAIN | VEGETABLES | DAIRY
+	slice_type = /obj/item/food/pizzaslice/mushroom
+
+/obj/item/food/pizza/mushroom/Initialize()
+	. = ..()
+	AddElement(/datum/element/processable, TOOL_KNIFE, /obj/item/food/pizzaslice/mushroom, 6, 30)
+
+/obj/item/food/pizza/mushroom/raw
+	name = "raw mushroom pizza"
+	icon_state = "mushroompizza_raw"
+	foodtypes =  GRAIN | VEGETABLES | DAIRY | RAW
+	burns_in_oven = FALSE
+	slice_type = null
+
+/obj/item/food/pizza/mushroom/raw/MakeBakeable()
+	AddComponent(/datum/component/bakeable, /obj/item/food/pizza/mushroom, rand(70 SECONDS, 80 SECONDS), TRUE, TRUE)
+
+/obj/item/food/pizzaslice/mushroom
+	name = "mushroom pizza slice"
+	desc = "Maybe it is the last slice of pizza in your life."
+	icon_state = "mushroompizzaslice"
+	tastes = list("crust" = 1, "tomato" = 1, "cheese" = 1, "mushroom" = 1)
+	foodtypes = GRAIN | VEGETABLES | DAIRY
+
+/obj/item/food/pizza/vegetable
+	name = "vegetable pizza"
+	desc = "No one of Tomatos Sapiens were harmed during making this pizza."
+	icon_state = "vegetablepizza"
+	food_reagents = list(/datum/reagent/consumable/nutriment = 25, /datum/reagent/consumable/tomatojuice = 6, /datum/reagent/medicine/oculine = 12, /datum/reagent/consumable/nutriment/vitamin = 5)
+	tastes = list("crust" = 1, "tomato" = 2, "cheese" = 1, "carrot" = 1)
+	foodtypes = GRAIN | VEGETABLES | DAIRY
+	slice_type = /obj/item/food/pizzaslice/vegetable
+
+/obj/item/food/pizza/vegetable/Initialize()
```

Comment:
Redundant due to the base typepath having a generic method, would actually create double.